main/src/ca/uwaterloo/flix/language/phase/Simplifier.scala

```diff
@@ -33,640 +32,611 @@ object Simplifier {
-            case TypeConstructor.False => MonoType.Unit
-            case TypeConstructor.Not => MonoType.Unit
-            case TypeConstructor.And => MonoType.Unit
-            case TypeConstructor.Or => MonoType.Unit
+          case TypeConstructor.Tuple(_) => MonoType.Tuple(args)
 
-            case TypeConstructor.Pure => MonoType.Unit
-            case TypeConstructor.EffUniv => MonoType.Unit
-            case TypeConstructor.Complement => MonoType.Unit
-            case TypeConstructor.Union => MonoType.Unit
-            case TypeConstructor.Intersection => MonoType.Unit
-            case TypeConstructor.Effect(_) => MonoType.Unit
-            case TypeConstructor.CaseSet(_, _) => MonoType.Unit
-            case TypeConstructor.CaseComplement(_) => MonoType.Unit
-            case TypeConstructor.CaseIntersection(_) => MonoType.Unit
-            case TypeConstructor.CaseUnion(_) => MonoType.Unit
+          case TypeConstructor.Arrow(_) => MonoType.Arrow(args.drop(1).init, args.last) // Erase the purity
 
-            case TypeConstructor.Relation =>
-              throw InternalCompilerException(s"Unexpected type: '$tpe'.", tpe.loc)
+          case TypeConstructor.RecordRowExtend(label) => MonoType.RecordExtend(label.name, args.head, args(1))
 
-            case TypeConstructor.Lattice =>
-              throw InternalCompilerException(s"Unexpected type: '$tpe'.", tpe.loc)
+          case TypeConstructor.Record => args.head
 
-            case TypeConstructor.SchemaRowEmpty =>
-              throw InternalCompilerException(s"Unexpected type: '$tpe'.", tpe.loc)
+          case TypeConstructor.True => MonoType.Unit
+          case TypeConstructor.False => MonoType.Unit
+          case TypeConstructor.Not => MonoType.Unit
+          case TypeConstructor.And => MonoType.Unit
+          case TypeConstructor.Or => MonoType.Unit
 
-            case TypeConstructor.SchemaRowExtend(_) =>
-              throw InternalCompilerException(s"Unexpected type: '$tpe'.", tpe.loc)
+          case TypeConstructor.Pure => MonoType.Unit
+          case TypeConstructor.EffUniv => MonoType.Unit
+          case TypeConstructor.Complement => MonoType.Unit
+          case TypeConstructor.Union => MonoType.Unit
+          case TypeConstructor.Intersection => MonoType.Unit
+          case TypeConstructor.Effect(_) => MonoType.Unit
+          case TypeConstructor.CaseSet(_, _) => MonoType.Unit
+          case TypeConstructor.CaseComplement(_) => MonoType.Unit
+          case TypeConstructor.CaseIntersection(_) => MonoType.Unit
+          case TypeConstructor.CaseUnion(_) => MonoType.Unit
 
-            case TypeConstructor.Schema =>
-              throw InternalCompilerException(s"Unexpected type: '$tpe'.", tpe.loc)
-          }
-      }
-    }
+          case TypeConstructor.Relation =>
+            throw InternalCompilerException(s"Unexpected type: '$tpe'.", tpe.loc)
 
-    /**
-      * Translates the given formal param `p` to the SimplifiedAst.
```

Comment:
Are we finally getting rid of the redundant documentation convention?

Reply:
Yeah, I think so, eventually. But we should have meaningful documentation. Somehow.